packages/components/src/select-control/index.js

```diff
@@ -217,63 +212,63 @@ export class SelectControl extends Component {
 	}
 
 	search( query ) {
-		const searchOptions = this.searchOptions || [];
-		const filteredOptions =
+		const cacheSearchOptions = this.cacheSearchOptions || [];
+		const searchOptions =
 			query !== null && ! query.length && ! this.props.hideBeforeSearch
-				? searchOptions
-				: this.getFilteredOptions( searchOptions, query );
+				? cacheSearchOptions
+				: this.getOptionsByQuery( cacheSearchOptions, query );
 
 		this.setState( {
 			query,
 			isFocused: true,
 			selectedIndex: 0,
-			filteredOptions,
-			isExpanded: Boolean( filteredOptions.length ),
+			searchOptions,
+			isExpanded: Boolean( searchOptions.length ),
```

Comment:
Because the filtering of excluded options no longer happens before this, this can set the `<List/>` as expanded even though no options will be shown.

To reproduce, search for products on the Products Overview, select them, and then repeat the search when all those options have been selected and note the empty list.

<img width="214" alt="Screen Shot 2021-01-25 at 12 33 42 PM" src="https://user-images.githubusercontent.com/10561050/105743990-a8215300-5f0a-11eb-9b27-58c3de473e52.png">
<img width="341" alt="Screen Shot 2021-01-25 at 12 32 46 PM" src="https://user-images.githubusercontent.com/10561050/105744177-bcfde680-5f0a-11eb-86c4-798b5a0fe555.png">

We could `return null;` if no options exist in the `List`, but I think since `isExpanded` controls some accessibility features, it might be better to filter the excluded options here.  Maybe we can create a method specifically for filtering out excluded options.  E.g., `getShownOptions()` or `filterExcludedOptions()`.

Reply:
Do you think this way would work?  I updated it so that if there are options, we setState i`sExpanded` to `true` and vise-versus.